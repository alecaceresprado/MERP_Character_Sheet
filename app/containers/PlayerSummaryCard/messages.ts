import { defineMessages } from 'react-intl';

export const scope = 'merp.HomePage';

export default defineMessages({
  viewButton: {
    id: `${scope}.view-button`,
    defaultMessage: 'View'
  },
  editButton: {
    id: `${scope}.edit-button`,
    defaultMessage: 'Edit'
  }
});
