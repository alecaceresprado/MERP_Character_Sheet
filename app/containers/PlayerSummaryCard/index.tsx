import React from 'react';
import Card from '@material-ui/core/Card';
import CardMedia from '@material-ui/core/CardMedia';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';
import CardActions from '@material-ui/core/CardActions';
import Button from '@material-ui/core/Button';
import { makeStyles } from '@material-ui/core/styles';
import Grid from '@material-ui/core/Grid';
import { FormattedMessage } from 'react-intl';

import { PlayerSummary } from 'models';
import messages from './messages';

const useStyles = makeStyles(() => ({
  card: {
    height: '100%',
    display: 'flex',
    flexDirection: 'column'
  },
  cardMedia: {
    paddingTop: '56.25%'
  },
  cardContent: {
    flexGrow: 1
  }
}));

export default function PlayerSummaryCard({ item }: { item: PlayerSummary }) {
  const classes = useStyles();

  return (
    <Grid item key={item.id} xs={12} sm={6} md={4}>
      <Card className={classes.card}>
        <CardMedia
          className={classes.cardMedia}
          image={item.avatarUrl}
          title="Image title"
        />
        <CardContent className={classes.cardContent}>
          <Typography gutterBottom variant="h5" component="h2">
            {item.name}
          </Typography>
          <Typography>{item.description}</Typography>
        </CardContent>
        <CardActions>
          <Button size="small" color="primary" href={`/player/${item.id}`}>
            <FormattedMessage {...messages.viewButton} />
          </Button>
          <Button size="small" color="primary" href={`/player/${item.id}/edit`}>
            <FormattedMessage {...messages.editButton} />
          </Button>
        </CardActions>
      </Card>
    </Grid>
  );
}
