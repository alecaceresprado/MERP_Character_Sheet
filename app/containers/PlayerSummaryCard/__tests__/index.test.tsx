import React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import 'jest-styled-components';

import PlayerSummaryCard from '../index';

const renderer = createRenderer();

const item = {
  id: '1',
  description: 'Braided long beard with charms with runes',
  avatarUrl:
    'https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcT3JvT2n7sHUoBw9QqyV-KlwYHfKphD4y0thI9k_iJsyeKmWwcZ&usqp=CAU',
  name: 'Lothar Picco'
};

describe('<PlayerSummaryCard />', () => {
  it('should render and match the snapshot', () => {
    renderer.render(<PlayerSummaryCard item={item} />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
