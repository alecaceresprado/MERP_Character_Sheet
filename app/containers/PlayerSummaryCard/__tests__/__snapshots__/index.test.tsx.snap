// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlayerSummaryCard /> should render and match the snapshot 1`] = `
<WithStyles(ForwardRef(Grid))
  item={true}
  md={4}
  sm={6}
  xs={12}
>
  <WithStyles(ForwardRef(Card))
    className="makeStyles-card-1"
  >
    <WithStyles(ForwardRef(CardMedia))
      className="makeStyles-cardMedia-2"
      image="https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcT3JvT2n7sHUoBw9QqyV-KlwYHfKphD4y0thI9k_iJsyeKmWwcZ&usqp=CAU"
      title="Image title"
    />
    <WithStyles(ForwardRef(CardContent))
      className="makeStyles-cardContent-3"
    >
      <WithStyles(ForwardRef(Typography))
        component="h2"
        gutterBottom={true}
        variant="h5"
      >
        Lothar Picco
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))>
        Braided long beard with charms with runes
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(CardContent))>
    <WithStyles(ForwardRef(CardActions))>
      <WithStyles(ForwardRef(Button))
        color="primary"
        href="/player/1"
        size="small"
      >
        <FormattedMessage
          defaultMessage="View"
          id="merp.HomePage.view-button"
          values={Object {}}
        />
      </WithStyles(ForwardRef(Button))>
      <WithStyles(ForwardRef(Button))
        color="primary"
        href="/player/1/edit"
        size="small"
      >
        <FormattedMessage
          defaultMessage="Edit"
          id="merp.HomePage.edit-button"
          values={Object {}}
        />
      </WithStyles(ForwardRef(Button))>
    </WithStyles(ForwardRef(CardActions))>
  </WithStyles(ForwardRef(Card))>
</WithStyles(ForwardRef(Grid))>
`;
